src/Microsoft.Health.Dicom.Core/Features/Store/IIndexDataStore.cs

```diff
@@ -139,20 +140,20 @@ public interface IIndexDataStore
     /// <summary>
     /// Asynchronously updates a DICOM instance NewWatermark
     /// </summary>
-    /// <param name="partitionKey">The partition key.</param>
+    /// <param name="partition">The partition.</param>
     /// <param name="versions">List of instances watermark to update</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>A task that with list of instance metadata with new watermark.</returns>
-    Task<IEnumerable<InstanceMetadata>> BeginUpdateInstanceAsync(int partitionKey, IReadOnlyCollection<long> versions, CancellationToken cancellationToken = default);
+    Task<IEnumerable<InstanceMetadata>> BeginUpdateInstanceAsync(Partition partition, IReadOnlyCollection<long> versions, CancellationToken cancellationToken = default);
```

Comment:
I am not sure why we needed this change?

Reply:
If that is the case, please pass the name why pass wrong value. You have the value in the FIlterAttribute, we just need to populate and pass on here